src/api/Controllers/ImportController.cs

```diff
@@ -84,33 +100,20 @@ public async Task<ActionResult<int>> UploadJsonFileAsync(int workgroupId, IFormF
             var user = await GetUserAsync().ConfigureAwait(false);
             var hydrotestCodelists = await GetHydrotestCodelistsAsync().ConfigureAwait(false);
 
-            foreach (var borehole in boreholes)
-            {
-                borehole.MarkAsNew();
-                borehole.Workgroup = null;
-                borehole.WorkgroupId = workgroupId;
-                borehole.LockedBy = null;
-                borehole.LockedById = null;
-                borehole.UpdatedBy = null;
-                borehole.CreatedBy = null;
+            MarkBoreholeContentAsNew(workgroupId, boreholes, user, hydrotestCodelists);
 
-                borehole.Stratigraphies?.MarkAsNew();
-                borehole.Completions?.MarkAsNew();
-                borehole.Sections?.MarkAsNew();
-                borehole.Observations?.MarkAsNew();
-                borehole.BoreholeGeometry?.MarkAsNew();
-
-                MapHydrotestCodelists(borehole, hydrotestCodelists);
+            await context.Boreholes.AddRangeAsync(boreholes).ConfigureAwait(false);
 
-                // Do not import any workflows from the json file but add a new unfinished workflow for the current user.
-                borehole.Workflows.Clear();
-                borehole.Workflows.Add(new Workflow { Role = Role.Editor, UserId = user.Id, Started = DateTime.Now.ToUniversalTime() });
+            if (FileTypeChecker.IsZip(boreholesFile))
+            {
+                // Save boreholes then upload attachments using the newly created borehole Ids
+                await context.SaveChangesAsync().ConfigureAwait(false);
+                await UploadAttachments(boreholesFile, boreholes).ConfigureAwait(false);
 
-                // Set the geometry's SRID to LV95 (EPSG:2056)
-                if (borehole.Geometry != null) borehole.Geometry.SRID = SpatialReferenceConstants.SridLv95;
+                // If any problem occurred while uploading the attachments, return a bad request.
+                if (!ModelState.IsValid) return ValidationProblem(statusCode: (int)HttpStatusCode.BadRequest);
```

Comment:
Q: Muss hier der `statusCode` explizit gesetzt werden?

Reply:
Leider ja, sonst bekommt man keinen Status Code zurück 🤷‍♀️

Reply:
Aber nur im Unit-Test, oder? Wenn man die Applikation laufen lässt, dann macht er es korrekt. Das war eine Erkenntnis von @Philippluca in einem anderen Projekt.

Reply:
Das ist sehr gut möglich, die dok der Method verspricht auch einen Badrequest. Fändest du es besser, wenn wir dann im Unit Test nicht auf Badrequest prüfen und stattdessen den Statuscode nicht explizit setzen?